3-UsingAPIs/Week2/homework/ex1-programmerFun/index.js

```diff
@@ -17,29 +17,36 @@ Full description at: https://github.com/HackYourFuture/Homework/blob/main/3-Usin
    url with `.shx`. There is no server at the modified url, therefore this 
    should result in a network (DNS) error.
 ------------------------------------------------------------------------------*/
-function requestData(url) {
-  // TODO return a promise using `fetch()`
+async function requestData(url) {
+  const response = await fetch(url);
+  if (response.ok) {
+    return await response.json();
```

Comment:
Now that you have removed the `try/catch` block you can return the promise from `response.json()` directly. Your code effectively does this:

```js
const data = await response.json();
return data;
```
Since an `async` function always returns a promise this code effectively returns a promise that resolves to the resolved value of `response.json()`.

But `response.json()` already returns a promise with that same value. Therefore you can just do this:

```suggestion
    return response.json();
```

No need to `await` here.